lib/api/controllers/adminController.js

```diff
@@ -0,0 +1,177 @@
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const
+  Bluebird = require('bluebird'),
+  {
+    BadRequestError,
+    KuzzleInternalError
+  } = require('kuzzle-common-objects').errors,
+  Request = require('kuzzle-common-objects').Request,
+  {
+    assertArgsHasAttribute
+  } = require('../../util/requestAssertions');
+
+/**
+ * @class AdminController
+ * @param {Kuzzle} kuzzle
+ */
+class AdminController {
+  constructor(kuzzle) {
+    this.kuzzle = kuzzle;
+  }
+
+  /**
+   * Reset the internal storage components (internalEngine index, cache and memory storage)
+   */
+  resetKuzzleData () {
+    this.kuzzle.pluginsManager.trigger('log:info', 'Kuzzle reset initiated: this may take a while...');
+
+    return deleteObjects(this.kuzzle, 'user', {})
+      .then(() => this.kuzzle.internalEngine.deleteIndex())
+      .then(() => this.kuzzle.pluginsManager.trigger('log:info', 'Kuzzle internal database deleted'))
+      .then(() => this.kuzzle.services.list.internalCache.flushdb())
+      .then(() => {
+        this.kuzzle.indexCache.remove(this.kuzzle.internalEngine.index);
+        this.kuzzle.pluginsManager.trigger('log:info', 'Kuzzle internal cache flushed');
+
+        return this.kuzzle.internalEngine.bootstrap.all();
+      });
+  }
+
+  /**
+   * Reset Redis cache
+   */
+  resetCache (request) {
+    assertArgsHasAttribute(request, 'database');
+
+    const
+      database = request.input.args.database,
+      cacheEngine = this.kuzzle.services.list[database];
+
+    if (! (cacheEngine !== undefined && typeof cacheEngine.flushdb === 'function')) {
+      throw new BadRequestError(`Database ${database} not found`);
+    }
+
+    return new Bluebird((resolve, reject) => cacheEngine.flushdb(error => error ? reject(error) : resolve()));
+  }
+
+  /**
+   * Reset all roles, profiles and users
+   */
+  resetSecurity () {
+    const options = {
+      refresh: 'wait_for'
+    };
+
+    return deleteObjects(this.kuzzle, 'user', options)
+      .then(() => deleteObjects(this.kuzzle, 'profile', options))
+      .then(() => deleteObjects(this.kuzzle, 'role', options));
+  }
+
+  /**
+   * Reset all indexes
+   */
+  resetDatabase () {
+    const indexes = Object.keys(this.kuzzle.indexCache.indexes).filter(idx => idx[0] !== '%');
+
+    return Bluebird.map(indexes, index => {
+      const request = new Request({ index });
+
+      return this.kuzzle.services.list.storageEngine.deleteIndex(request)
+        .then(() => delete this.kuzzle.indexCache.indexes[index]);
+    });
+  }
+
+  /**
+   * Shutdown Kuzzle
+   */
+  shutdown () {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this function is empty or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1186)